Scripts/Mend SCA/group-permissions.sh

```diff
@@ -0,0 +1,88 @@
+# MEND_ORG_UUID - optional for selecting a different organization
+
+group_name=$1
+group_role=$2
+
+
+
+function login() {
+    MEND_API_URL="$(echo "${MEND_URL}" | sed -E 's/(saas|app)(.*)/api-\1\2\/api\/v2.0/g')"
+
+    if [ -z "${MEND_EMAIL}" ]; then
+      echo "MEND_EMAIL is not set. Please set it before proceeding."
+      exit 1
+    fi
+    
+    if [ -z "${MEND_USER_KEY}" ]; then
+      echo "MEND_USER_KEY is not set. Please set it before proceeding."
+      exit 1
+    fi
+
+    if [ -z "${MEND_URL}" ]; then
+      echo "MEND_URL is not set. Please set it before proceeding."
+      exit 1
+    fi
+
+    if [ -z "$MEND_ORG_UUID" ]; then
+        echo "MEND_ORG_UUID environment variable was not provided."
+        echo -e "The Login API will default to the last organization this user accessed in the MEND UI.\n"
+        LOGIN_BODY="{\"email\": \"$MEND_EMAIL\", \"userKey\": \"$MEND_USER_KEY\" }"
+    else
+        echo -e "Logging in with provided API key.\n"
+        LOGIN_BODY="{\"email\": \"$MEND_EMAIL\", \"userKey\": \"$MEND_USER_KEY\", \"orgToken\": \"$MEND_ORG_UUID\"}"
+    fi
+
+    # Log into API 2.0 and get the JWT Token, Organization UUID, and Organization Name
+    LOGIN_RESPONSE=$(curl -s -X POST --location "$MEND_API_URL/login" --header 'Content-Type: application/json' --data-raw "${LOGIN_BODY}")
+
+    JWT_TOKEN=$(echo "$LOGIN_RESPONSE" | jq -r '.retVal.jwtToken')
+    MEND_ORG_UUID=$(echo "$LOGIN_RESPONSE" | jq -r '.retVal.orgUuid')
+    MEND_ORG_NAME=$(echo "$LOGIN_RESPONSE" | jq -r '.retVal.orgName')
+}
+
+function find_group() {
+    GROUPS_RESPONSE=$(curl -s --location "$MEND_API_URL/orgs/$MEND_ORG_UUID/groups" --header "Content-Type: application/json" --header "Authorization: Bearer $JWT_TOKEN")
+
+    if [ -z "$group_name" ]; then
+        echo "Please add a group name from the list when calling the script."
+        echo "Example:  ./group-permissions.sh mygroupname"
+        echo "$GROUPS_RESPONSE" | jq -r '.retVal[] | {name: .name, uuid: .uuid}'
+        exit 1
+    else
+        set_permissions
+    fi
+    
+    
+}
+
+function set_permissions() {
+    GROUP_UUID=$(echo "$GROUPS_RESPONSE" | jq --arg name_to_find $group_name -r '.retVal[] | select(.name == $name_to_find) .uuid')
+    if [ -z "$group_role" ]; then
+        echo "group_role not set, USER role will be used by default, set a desired role other than USER as the 2nd variable"
+        echo "https://docs.mend.io/bundle/mend-api-2-0/page/index.html#tag/User-Management-Groups/operation/addGroupRoles"
+        group_role="USER"
+        
+    fi
+
+    echo "Adding organization level $group_role permissions to group $group_name with uuid of $GROUP_UUID"
+    ADDROLE_BODY="{\"contextType\": \"orgs\", \"contextToken\": \"$MEND_ORG_UUID\", \"role\": \"$group_role\"}"
+    curl -s --location "$MEND_API_URL/orgs/$MEND_ORG_UUID/groups/$GROUP_UUID/roles" --header "Content-Type: application/json" --header "Authorization: Bearer $JWT_TOKEN" -d "${ADDROLE_BODY}"
```

Comment:
Should send this to a variable like ``role_response=...``
Then a comparison should be made for the output.
```sh
response_status=$(echo $role_response | jq -re '.status')
if [[ $response_status -ne "null" ]]; then
  echo $role_response
fi
```

This will allow the user to see the result of this API call when any error occurs (the "status" key is only present in the response if an error occurs). Without this change made, the output of the API will always be printed.

Reply:
Wouldn't an end user always want the result to be printed regardless of success or failure so that they could see what the API call resulted in?

Reply:
Not necessarily in a JSON format with all of the information that is actually given. Maybe could add an else statement that tells it to echo "Role assignment complete" or something to that effect. At the very least if you believe we should actually output all information, then I would say pipe the output to jq just so that it pretty prints it.